### DISCUSSION TITLE
output from prediction.rfsrc

### DISCUSSION THREAD

**Discussion by petterlindgrensogeti** [kogalur/randomForestSRC#120] [Q&A] [answered]
Hi, I am running rfsrc for predicting length of stay at for patients at a hospital. 

Is it possible the get the true error rate in predict.rfsrc? As I understand the documentation it is the OOB Error Rate. I thought OOB Error rate is good when you are training but when I want to test my model on unseen data I want the true error rate.

Also, when do you need the option outcome="test" in predict? I want the predictions do be based on my tree built in train, not adjusted by y-outcome in test.

And lastly, is it possible to get Brier score from a prediction run?  I would like brier score to see how good my model is on unseen data

**Comment by ishwaran** [comment, marked as answer]
The OOB error rate is the leave-one-out (CV) estimate of prediction error.  This is the correct value for assessing prediction performance on unseen data.  

1. Note you can always obtain a traditional CV error estimate, such as  K-fold validation, but this will require repeated training/testing and therefore running several forest runs.  The OOB error rate is computationally fast to get and also is a smoothed CV(1) estimate and therefore has good properties.  Generally it matches very well what you would obtain using a true hold out data set.

2. When it comes to inference for the training data (which is somewhat related to what you are asking), the OOB ensemble is again the right estimator to use.  Note that the OOB error rate is obtained from the OOB ensemble.

3. The outcome = "test" in predict is used when you want to run test data down the trained forest, replacing the terminal node estimator with the estimator calculated using the new test data as opposed to the training data.  This is very specialized and if you don't fully understand this, then this is probably not the option for you.

Regarding the Brier score.  Yes, the new release 2.14.0 has the ability, through the new function "get.brier.survival".  See the vignette "Random Survival Forests" and the section "Plotting the Brier Score and CRPS" on our webpage:

https://luminwin.github.io/randomForestSRC/articles/survival.html
